src/main/java/com/ita/if103java/ims/dao/impl/AccountDaoImpl.java

```diff
@@ -0,0 +1,185 @@
+        }
+        if (status == 0)
+            throw accountEntityNotFoundException("Update account exception", "id = " + account.getId());
+
+        return account;
+    }
+
+    @Override
+    public boolean activate(Long id) {
+        int status;
+        try {
+            status = jdbcTemplate.update(Queries.SQL_SET_ACTIVE_STATUS_ACCOUNT, true, id);
+
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "activate", "id = " + id);
+        }
+        if (status == 0)
+            throw accountEntityNotFoundException("Activate account exception", "id = " + id);
+
+        return true;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        int status;
+        try {
+            status = jdbcTemplate.update(Queries.SQL_SET_ACTIVE_STATUS_ACCOUNT, false, id);
+
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "delete", "id = " + id);
+        }
+        if (status == 0)
+            throw accountEntityNotFoundException("Delete account exception", "id = " + id);
+
+        return true;
+    }
+
+    @Override
+    public boolean updateToPremium(Long id) {
+        int status;
+        try {
+            status = jdbcTemplate.update(Queries.SQL_UPDATE_ACCOUNT_TO_PREMIUM, id);
+
+        } catch (DataAccessException e) {
+            throw crudException(e.getMessage(), "update", "id = " + id);
+        }
+        if (status == 0)
+            throw accountEntityNotFoundException("Update account to Premium exception", "id = " + id);
+
+        return true;
+    }
+
+    private PreparedStatement createStatement(Account account, ZonedDateTime currentDateTime, Connection connection) throws SQLException {
+        PreparedStatement preparedStatement = connection.prepareStatement(Queries.SQL_CREATE_ACCOUNT, Statement.RETURN_GENERATED_KEYS);
+
+        int i = 0;
+        preparedStatement.setString(++i, account.getName());
+        preparedStatement.setObject(++i, account.getType());
+        preparedStatement.setLong(++i, account.getAdminId());
+        preparedStatement.setObject(++i, currentDateTime.toLocalDateTime());
+        preparedStatement.setBoolean(++i, account.isActive());
+
+
+        return preparedStatement;
+    }
+
+    private EntityNotFoundException accountEntityNotFoundException(String message, String attribute) {
+        EntityNotFoundException exception = new EntityNotFoundException(message);
+        logger.error("EntityNotFoundException exception. Account is not found ({}). Message: {}", attribute, message);
+        return exception;
+    }
+
+    private CRUDException crudException(String message, String operation, String attribute) {
+        CRUDException exception = new CRUDException(message);
+        logger.error("CRUDException exception. Operation:({}) account ({}) exception. Message: {}", operation, attribute, message);
+        return exception;
+    }
```

Comment:
Please sit together with the team and think how to move this similar methods to some abstract class and generalize them in order they can be accessed by every DAO